File: JAVA_CalculadoraReflection/src/br/ifes/projetosistemas/calculadora/view/Menu.java
package br.ifes.projetosistemas.calculadora.view;

import br.ifes.projetosistemas.calculadora.dto.RequestDTO;
import br.ifes.projetosistemas.calculadora.dto.ResponseDTO;

import java.util.Scanner;

public class Menu {

    public RequestDTO show(){
        Scanner input = new Scanner(System.in);
        System.out.println("#    JAVA  CALC    #");
        System.out.println("Escolha uma das opções:");
        System.out.println("1 - Somar");
        System.out.println("2 - Subtrair");
        System.out.println("3 - Multiplicar");
        System.out.println("4 - Dividir");
        System.out.println("5 - SAIR");

        int opcao = input.nextInt();

        System.out.println("Informe o primeiro valor:");
        int valor1 = input.nextInt();
        System.out.println("Informe o segundo valor:");
        int valor2 = input.nextInt();

        return new RequestDTO(opcao,valor1,valor2);

    }

    public void showResult(ResponseDTO responseDTO) {
        if (responseDTO.getResult() == -1) {
            System.out.println("Erro ao calcular.");
        } else {
            System.out.println(STR."O Resultado é: \{responseDTO.getResult()}");
        }
    }
}

File: CSharp_Calculadora/CalculadoraConsole/View/Menu.cs
using CalculadoraConsole.Controller;
using CalculadoraConsole.DTOs;
using System;

namespace CalculadoraConsole.View
{
    public class Menu
    {
        public static void Show()
        {
            var controller = new CalculadoraController();
            
            Console.WriteLine("Escolha uma das opções:");
            Console.WriteLine("# Somar");
            Console.WriteLine("# Subtrair");
            Console.WriteLine("# Multiplicar");
            Console.WriteLine("# Dividir");
            Console.WriteLine("# SAIR");

            string operacaoEscolhida = Console.ReadLine();

            Console.WriteLine("Digite o primeiro número:");
            int numero1 = Convert.ToInt16(Console.ReadLine());

            Console.WriteLine("Digite o segundo número:");
            int numero2 = Convert.ToInt16(Console.ReadLine());

            // Cria o DTO de request (entrada)
            var request = new OperacaoRequestDTO()
            {
                Numero1 = numero1,
                Numero2 = numero2,
                TipoOperacao = operacaoEscolhida
            };

            // Chama o controller e obtém o DTO de response (saída)
            var response = controller.ExecutarOperacao(request);

            // Exibe o resultado
            Console.WriteLine(response.Mensagem);
            if (response.Mensagem.Contains("sucesso"))
            {
                Console.WriteLine($"Resultado: {response.Resultado}");
            }
        }
    }
}